tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/mult/MultiplyRight.java

```diff
@@ -0,0 +1,158 @@
+package dk.alexandra.fresco.tools.mascot.mult;
+
+import dk.alexandra.fresco.framework.network.Network;
+import dk.alexandra.fresco.framework.util.Pair;
+import dk.alexandra.fresco.framework.util.StrictBitVector;
+import dk.alexandra.fresco.tools.mascot.MascotResourcePool;
+import dk.alexandra.fresco.tools.mascot.field.FieldElement;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Actively-secure two-party protocol for computing the product of two secret inputs.
+ *
+ * <p>One input is held by the left party, the other by the right party. The protocol is asymmetric
+ * in the sense that the left party performs a different computation from the right party. This
+ * class implements the functionality of the right party. For the other side,
+ * see {@link MultiplyLeft}. The resulting product is secret-shared among the two parties.</p>
+ */
+public class MultiplyRight extends MultiplyShared {
+
+  public MultiplyRight(MascotResourcePool resourcePool, Network network, Integer otherId,
+      int numLeftFactors) {
+    super(resourcePool, network, otherId, numLeftFactors);
+  }
+
+  public MultiplyRight(MascotResourcePool resourcePool, Network network, Integer otherId) {
+    this(resourcePool, network, otherId, 1);
+  }
+
+  /**
+   * Generate random seed pairs using OT. <br>
+   * The seed pairs are correlated with the multiplication factors of the other party. If the other
+   * party's factors (represented as a bit vector) is 010, this party will receive seed pairs
+   * <i>(a<sub>0</sub>, a<sub>1</sub>), (b<sub>0</sub>, b<sub>1</sub>), (c<sub>0</sub>,
+   * c<sub>1</sub>)</i> whereas the other party will receive seeds <i>a<sub>0</sub>, b<sub>1</sub>,
+   * c<sub>0</sub></i>. The parties can use the resulting seeds to compute the shares of the product
+   * of their factors.
+   *
+   * @param numMults the number of total multiplications
+   * @param seedLength the bit length of the seeds
+   * @return the seed pairs
+   */
+  public List<Pair<StrictBitVector, StrictBitVector>> generateSeeds(int numMults, int seedLength) {
```

Comment:
Its a little confusing to call these "seeds". They are not really seeds (we use them as seeds in the COPE protocol, but not the triplegeneration). GetOtMessages, would probably be more correct.